airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java

```diff
@@ -0,0 +1,139 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnowflakeDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeDestination.class);
+
+  protected static final String COLUMN_NAME = "data";
+
+  @Override
+  public ConnectorSpecification spec() throws IOException {
+    final String resourceString = MoreResources.readResource("spec.json");
+    return Jsons.deserialize(resourceString, ConnectorSpecification.class);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      final Supplier<Connection> connectionFactory = SnowflakeDatabase.getConnectionFactory(config);
+      SnowflakeDatabase.executeSync(connectionFactory, "SELECT 1;");
+      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+    } catch (Exception e) {
+      return new AirbyteConnectionStatus().withStatus(Status.FAILED).withMessage(e.getMessage());
+    }
+  }
+
+  /**
+   * Strategy:
+   * <p>
+   * 1. Create a temporary table for each stream
+   * </p>
+   * <p>
+   * 2. Accumulate records in a buffer. One buffer per stream.
+   * </p>
+   * <p>
+   * 3. As records accumulate write them in batch to the database. We set a minimum numbers of records
+   * before writing to avoid wasteful record-wise writes.
+   * </p>
+   * <p>
+   * 4. Once all records have been written to buffer, flush the buffer and write any remaining records
+   * to the database (regardless of how few are left).
+   * </p>
+   * <p>
+   * 5. In a single transaction, delete the target tables if they exist and rename the temp tables to
+   * the final table name.
+   * </p>
+   *
+   * @param config - Snowflake-specific configuration object as json.
+   * @param catalog - schema of the incoming messages.
+   * @return consumer that writes singer messages to the database.
+   * @throws Exception - anything could happen!
+   */
+  @Override
+  public DestinationConsumer<AirbyteMessage> write(JsonNode config, AirbyteCatalog catalog) throws Exception {
+    // connect to snowflake
+    final Supplier<Connection> connectionFactory = SnowflakeDatabase.getConnectionFactory(config);
+    Map<String, SnowflakeWriteConfig> writeBuffers = new HashMap<>();
+
+    // create transient tables if they do not exist
+    // we use transient instead of temporary because all of our statements don't run in a single session
+    for (final AirbyteStream stream : catalog.getStreams()) {
+      final String tableName = stream.getName();
+      final String tmpTableName = stream.getName() + "_" + Instant.now().toEpochMilli();
+
+      final String query = String.format(
+          "CREATE TRANSIENT TABLE IF NOT EXISTS \"%s\" ( \n"
+              + "\"ab_id\" VARCHAR PRIMARY KEY,\n"
+              + "\"%s\" VARIANT,\n"
+              + "\"emitted_at\" TIMESTAMP WITH TIME ZONE DEFAULT current_timestamp()\n"
```

Comment:
Shoud it be the current_timestamp or the one in the record?

Reply:
that's right, fixed.